### PR TITLE
Added unit tests for issue #339

### PR DESCRIPTION
**Pull request by charjhall**
Modifies MockSemanticTree to increase the depth of the tree and adds two unit tests for issue #339

---
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/thecombine/392)

### CODE REVIEW

File: src/components/TreeView/tests/TreeViewHeader.test.tsx [hunk 1]
```diff
@@ -74,6 +74,33 @@ describe("Tests TreeViewHeader", () => {
     expect(event.target.value).toEqual(TEST);
   });
 
+  it("Search & select domain switches on a length 5 number", () => {
+    var leafnode: SemanticDomainWithSubdomains =
```
Comment (johnthagen) [inline src/components/TreeView/tests/TreeViewHeader.test.tsx:78]: Can this be changed to `const leafnode` to make the binding immutable?

Comment (charjhall) [reply src/components/TreeView/tests/TreeViewHeader.test.tsx:78]: Yes most certainly. I was actually unsure of whether to declare this as a const, var, or let. Is it good practice to declare const if it is not changing?

Comment (johnthagen) [reply src/components/TreeView/tests/TreeViewHeader.test.tsx]: Yes, in modern JS/TS you should default to always using `const` (though this code base could be cleaned up a bit in that regard). This makes the binding immutable and can avoid subtle errors of variables being updated or reassigned unintentionally.

If the value needs to be changed, use `let`.

`var` should be avoided in most cases.

Comment (johnthagen) [reply src/components/TreeView/tests/TreeViewHeader.test.tsx]: Also, JS/TS conventions are to use `lowerCamelCase`, so style-wise I think the best result is:

```typescript
const leafNode: ...
```

**Comment by johnthagen** [comment]
@charjhall Just an FYI, due to #320 we are on a somewhat older version of Jest and such. So if you run into things you see online that are different to how they can be done here, that might be part of the reason.

**Pull request merged**